### PR TITLE
add listener fn in groupsService

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/components/groups/groupsService.js [hunk 1]
```diff
@@ -58,23 +61,12 @@
...
-                    var getAllGroupsProm =  _getStorage().get(_getGroupPath()).then(function (groups) {
-                        if(angular.equals(groups, {})){
-                            groups[GroupsService.defaultGroupName] = {
-                                name: GroupsService.defaultGroupName,
-                                groupKey: GroupsService.defaultGroupName
-                            };
-                        }
-                        GroupsService.groups = groups;
+                GroupsService.getAllGroups = function () {
+                    return storageSrv.get(GROUPS_PATH, groupsDefault()).then(function (groups) {
```
Comment (assafshp) [inline src/components/groups/groupsService.js]: why did you change that ?

File: src/components/groups/groupsService.js [hunk 2]
```diff
@@ -140,47 +132,41 @@
...
                     }
-                }
+                };
 
-                function groupsChildAdded(dataSnapshot) {
-                    $timeout(function () {
-                        GroupsService.groups[dataSnapshot.key()] = dataSnapshot.val();
-                    });
-                }
+                GroupsService.removeGroupsListener = function (eventName, callback) {
```
Comment (assafshp) [inline src/components/groups/groupsService.js]: addGroupsListener and remove function are the same.You can use one func with param

File: src/components/groups/groupsService.js (hunk 1 above)
Comment (orenor10) [inline src/components/groups/groupsService.js]: the deleted code handle the default value for get.
In the new code the default value return from groupsDefault() (line 65)

File: src/components/groups/groupsService.js (hunk 2 above)
Comment (orenor10) [inline src/components/groups/groupsService.js]: done
